### PR TITLE
New news

### PR DESCRIPTION
**Pull request by lucasvanhalst**

### CODE REVIEW

File: src/rogerthat/bizz/news/__init__.py [hunk 1]
```diff
@@ -1063,13 +1072,16 @@ def trans(news_item_id, news_type):
                 news_item.qr_code_content = qr_code_content
         if media_changed:
             news_item.media = media and NewsMedia(content=news_item_image_id,
+                                                  url=media.content if isinstance(media.content, basestring) else None,
```
Comment (RubenMattan) [inline src/rogerthat/bizz/news/__init__.py]: url=None if news_item_image_id  else media.content

File: src-test/rogerthat_tests/mobicage/service/test_news.py [hunk 2]
```diff
@@ -604,20 +604,20 @@ def test_news_locations(self):
                 m = _create_news_item_match(s, news_item_0)
                 self.assertIsNotNone(m)
 
-
             news_item_1_locations = NewsLocationsTO()
             news_item_1_locations.match_required = False
             news_item_1_locations.geo_addresses = [NewsGeoAddressTO(distance=ni3_distance,
-                                                                    latitude=long(ni3_lat * 1000000),
-                                                                    longitude=long(ni3_lng * 1000000))]
+                                                                    latitude=ni3_lat,
```
Comment (RubenMattan) [inline src-test/rogerthat_tests/mobicage/service/test_news.py]: revert using floats gives issues (precision, ...)

File: src-test/rogerthat_tests/mobicage/service/test_news.py [hunk 3]
```diff
@@ -647,7 +647,7 @@ def test_news_locations(self):
 
             self.assertEqual(news_item_2.has_locations, True)
             self.assertEqual(news_item_2.location_match_required, True)
-            self.assertEqual(len(news_item_2.locations), 1)
+            self.assertEqual(len(news_item_2.locations.addresses), 1)
```
Comment (RubenMattan) [inline src-test/rogerthat_tests/mobicage/service/test_news.py]: add `self.assertEqual(len(news_item_2.locations.geo_addresses), 0)`

File: src-test/rogerthat_tests/mobicage/service/test_news.py [hunk 4]
```diff
@@ -665,8 +665,8 @@ def test_news_locations(self):
             news_item_3_locations = NewsLocationsTO()
             news_item_3_locations.match_required = True
             news_item_3_locations.geo_addresses = [NewsGeoAddressTO(distance=ni1_distance,
-                                                                    latitude=long(ni1_lat * 1000000),
-                                                                    longitude=long(ni1_lng * 1000000))]
+                                                                    latitude=ni1_lat,
```
Comment (RubenMattan) [inline src-test/rogerthat_tests/mobicage/service/test_news.py]: revert using floats gives issues (precision, ...)

File: src/rogerthat/bizz/jobs/__init__.py [hunk 5]
```diff
@@ -157,8 +147,7 @@ def save_job_criteria(app_user, location, contract_types, job_domains, keywords,
 
     if location.geo:
         should_build_matches = True
-        job_criteria.geo_location = ndb.GeoPt(location.geo.latitude / 1000000.0,
-                                              location.geo.longitude / 1000000.0)
+        job_criteria.geo_location = ndb.GeoPt(location.geo.latitude, location.geo.longitude)
```
Comment (RubenMattan) [inline src/rogerthat/bizz/jobs/__init__.py]: revert using floats gives issues (precision, ...)

File: src/rogerthat/bizz/location.py [hunk 6]
```diff
@@ -123,7 +123,7 @@ def encode_request():
             raise CannotDetermineLocationException()
 
         (_, _, _, latitude, longitude, _, _, _) = unpack(">hBiiiiih", response.content)
-        return _to_geo_point_to(db.GeoPt(latitude / 1000000.0, longitude / 1000000.0))
+        return _to_geo_point_to(db.GeoPt(latitude, longitude))
```
Comment (RubenMattan) [inline src/rogerthat/bizz/location.py]: revert using floats gives issues (precision, ...)

File: src/rogerthat/bizz/location.py [hunk 7]
```diff
@@ -193,8 +193,8 @@ def parse_location(friend_language, accuracy, geocoded_results):
     loc.accuracy = location.accuracy
     loc.timestamp = timestamp
 
-    maps_lat = loc.latitude / 1000000.0
-    maps_long = loc.longitude / 1000000.0
+    maps_lat = loc.latitude
```
Comment (RubenMattan) [inline src/rogerthat/bizz/location.py]: revert using floats gives issues (precision, ...)

File: src/rogerthat/bizz/location.py [hunk 8]
```diff
@@ -140,7 +140,7 @@ def encode_request():
     ct.lac = raw_info.lac
     ct.cid = raw_info.cid
     ct.net = raw_info.net
-    ct.geoPoint = db.GeoPt(geoPointTO.latitude / 1000000.0, geoPointTO.longitude / 1000000.0)
+    ct.geoPoint = db.GeoPt(geoPointTO.latitude , geoPointTO.longitude )
```
Comment (RubenMattan) [inline src/rogerthat/bizz/location.py]: revert using floats gives issues (precision, ...)

File: src/rogerthat/bizz/news/__init__.py [hunk 9]
```diff
@@ -1125,30 +1125,24 @@ def trans(news_item_id, news_type):
...
-                    l.address.address_uid = UserProfileInfoAddress.create_uid([l.address.country_code,
-                                                                               l.address.zip_code,
-                                                                               l.address.street_name])
-
-                    news_item.locations.append(l)
+                    if address.level != NewsItemAddress.LEVEL_STREET:
+                        raise BusinessException('Invalid address level: %s' % address.level)
+                news_item.locations = NewsItemLocation(
+                    geo_addresses=[NewsItemGeoAddress(
+                        geo_location=ndb.GeoPt(geo_address.latitude, geo_address.longitude),
```
Comment (RubenMattan) [inline src/rogerthat/bizz/news/__init__.py]: revert using floats gives issues (precision, ...)

File: src/rogerthat/bizz/news/matching.py [hunk 10]
```diff
@@ -363,27 +363,25 @@ def _match_roles_of_item(app_user, news_item):
...
-        if l.geo_address:
-            for a in upi.addresses:
-                if should_match_location(l.geo_address.geo_location.lat,
-                                         l.geo_address.geo_location.lon,
-                                         l.geo_address.distance,
-                                         a.geo_location.lat,
-                                         a.geo_location.lon,
-                                         a.distance):
+
+    if news_item.locations:
```
Comment (RubenMattan) [inline src/rogerthat/bizz/news/matching.py]: if not ... return False

this way its clearer

File: src/rogerthat/to/news.py [hunk 11]
```diff
@@ -249,8 +248,14 @@ class MediaTO(BaseMediaTO):
 
 class NewsGeoAddressTO(TO):
     distance = long_property('1')
-    latitude = long_property('2')
-    longitude = long_property('3')
+    latitude = float_property('2')
+    longitude = float_property('3')
```
Comment (RubenMattan) [inline src/rogerthat/to/news.py]: use longs

Comment (lucasvanhalst) [reply src/rogerthat/to/news.py]: why though, we use floats for location in location widget / form too

Comment (lucasvanhalst) [reply src/rogerthat/to/news.py]: Datestore also saves them as floats `datastore_types.GeoPt(51.09222495439189, 3.8203423860190924)`

Comment (RubenMattan) [reply src/rogerthat/to/news.py]: We always use long for locations, having both is just confusing

Comment (RubenMattan) [reply src/rogerthat/to/news.py]: Its not that consistent as i thought ...
@bart-at-mobicage longs or floats?

**Pull request merged**